### DISCUSSION TITLE
[Box] Add bordered prop

### DISCUSSION THREAD

**Discussion by aweell** [dxc-technology/halstack-react#939] [General] [answered]
## Component API proposal

| Prop | Default | Description |
| :---- | :---------- | :--------|
| `borderStyle` | `none` | Can be defined as `none`, `solid` or `dashed` |
| `shadow` | `none` | Can be defined as `low`, `medium` or `high` |
| `padding` | `none` | Can be defined as `xsmall`, `small`, `medium`, `large` and `xlarge` |
| `margin` | `none` | Can be defined as `xsmall`, `small`, `medium`, `large` and `xlarge` |
| `display` | `inline-flex` | Change the display of the box component |

<img width="1154" alt="Screen Shot 2022-04-29 at 9 06 03 AM" src="https://user-images.githubusercontent.com/44420072/165899161-77470163-fae6-4265-b36d-644ac0f58a91.png">

Fig. 1 - Examples of `borderStyle="solid"`, `shadow="low"`, and `borderStyle="dashed"`

**Comment by GomezIvann** [comment]
I want to bring some topics to the table:

- Would it be interesting to add the `as` prop?
- If we open the door to border styling, should we also add a `borderRadius` prop?
- `width` and `height` could be potentially useful props, or should the box autosize based on the layout components that contain it?
- Should we add any kind of mechanism to align the content inside the box?

**Comment by GomezIvann** [comment, marked as answer]
I mark this discussion as answered since we decided some time ago to rethink and reimplement this component (https://github.com/dxc-technology/halstack-react/issues/1213) as our plan of offering Layout components to the users so they can create consistent and semantic layouts using Halstack DS.
